src/python/WMCore/Database/DBCore.py

```diff
@@ -96,9 +102,10 @@ def executemanybinds(self, s=None, b=None, connection=None,
             Trying to select many
             """
             if returnCursor:
-                result = []
-                for bind in b:
-                    result.append(connection.execute(s, bind))
+                if b:
+                    return self._generateCursor(s, b, connection)
```

Comment:
Why a helper function here?  `executemanybinds` isn't exactly overly-long and it seems to be the only user of this two-line method.

Reply:
It seems it doesn't allow return and yield within the same function. (Gives me syntax error) So I had to add a separate helper function.

Reply:
Hm.  Might be better to consistently yield a generator from here.  Mixing generators and values is a fairly confusing call interface.  How about always using generators?  In that case, we'd have:

```
>>> def test(foo):
...   if foo:
...     for f in foo:
...       yield f
...   else:
...     raise StopIteration()
... 
>>> for t in test([]): print t
... 
>>> for t in test([1,2,3,4]): print t,
... 
1 2 3 4
>>> print test([])
<generator object test at 0x10226eb40>
>>> print test([1,2,3,4])
<generator object test at 0x10226eb40>
```

Reply:
> Hm. Might be better to consistently yield a generator from here. Mixing generators and values is a fairly confusing call interface. How about always using generators? In that case, we'd have:

I added a few comments below regarding this. (Basically more changes need to be made, including outside WMCore(i.e. DBS) :-))
I will check how bad it is to change this.